### PR TITLE
Add support for OAuth2 authentication

### PR DESCRIPTION
**Pull request by selimnairb**

### CODE REVIEW

**Comment by selimnairb** [comment]
@pkdash @danames Can you review this PR?

File: README.rst [hunk 1]
```diff
@@ -29,13 +29,86 @@ To authenticate, and then get a list of resources you have access to::
...
+
+To authenticate using OAuth2 authentication (using an existing token), and then get a list of resources you have
+access to::
+
+    from oauthlib.oauth2 import TokenExpiredError
+    from hs_restclient import HydroShare, HydroShareAuthOAuth2
+
+    # Get a client ID and client secret by registering a new application at:
+    # https://www.hydroshare.org/o/applications/
+    # Choose client type "Confidential" and authorization grant type "Resource owner password-based"
```
Comment (selimnairb) [inline README.rst]: @danames @jsadler2 Is this the correct grant type for the Tethys application?  I'm not sure what you are using.

**Comment by selimnairb** [comment]
@pkdash Can you please review.  I'd like to get this released this week.

File: README.rst [hunk 2]
```diff
@@ -29,13 +29,86 @@ To authenticate, and then get a list of resources you have access to::
...
+
+    # A token dictionary obtained separately from HydroShare of the form:
+    #   {
+    #       "access_token": "<your_access_token>",
+    #       "token_type": "Bearer",
+    #       "expires_in": 36000,
+    #       "refresh_token": "<your_refresh_token>",
+    #       "scope": "read write groups"
+    #   }
+    token = get_token()
```
Comment (pkdash) [inline README.rst]: ```get_token()``` comes from which module?

**Comment by pkdash** [comment]
@selimnairb I have made one comment. Otherwise it looks good even though I don't know much about Oauth.

File: README.rst (hunk 2 above)
Comment (selimnairb) [inline README.rst]: It's meant to be an example of a function in their code.  I will make a note of this so it is clear.

**Comment by pkdash** [comment]
+1

**Pull request merged**